packages/backend-common/src/util/KubernetesContainerRunner.ts

```diff
@@ -370,9 +383,16 @@ export class KubernetesContainerRunner implements ContainerRunner {
   }
 }
 
-function handleKubernetesError(message: string, err: Error): Error {
-  if (err instanceof HttpError) {
-    return new Error(`${message} ${(err.body as V1Status).message}`);
+function handleKubernetesError(message: string, err: any): Error {
+  if (err && typeof err.body === 'string') {
+    try {
+      const parsedBody = JSON.parse(err.body);
+      if (parsedBody && parsedBody.message) {
+        return new Error(`${message} ${parsedBody.message}`);
+      }
+    } catch (e) {
+      return new Error(`${message} Error parsing response body - ${e.message}`);
```

Comment:
maybe you could avoid throwing this error so that you have a more descriptive error using the fallback below?

```suggestion
```

Reply:
Good Idea ;)